packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -88,29 +88,29 @@ export class CoffeeCompiler extends CachingCompiler {
       inputFile.getDeclaredExports().map(e => e.name)
     );
 
-    if (/`|\b(?:import|export|yield)\b/.test(source)) {
-      // If source contains backticks or features that output as ES2015+,
-      // pass the coffee output through babel-compiler
-      const doubleRoastedCoffee =
-        this.babelCompiler.processOneFileForTarget(inputFile, output.js);
-
-      if (doubleRoastedCoffee != null &&
-          doubleRoastedCoffee.data != null) {
-        output.js = doubleRoastedCoffee.data;
-
-        if (doubleRoastedCoffee.sourceMap) {
-          // Combine the original CoffeeScript source map with the one
-          // produced by this.babelCompiler.processOneFileForTarget.
-          const smg = new SourceMapGenerator(
-            new SourceMapConsumer(doubleRoastedCoffee.sourceMap));
-          smg.applySourceMap(new SourceMapConsumer(sourceMap));
-          sourceMap = smg.toJSON();
-        } else {
-          // If the .coffee file is contained by a node_modules directory,
-          // then BabelCompiler will not transpile it, and there will be
-          // no sourceMap, but that's fine because the original
-          // CoffeeScript sourceMap will still be valid.
-        }
+    // CoffeeScript contains a handful of features that output as ES2015+,
+    // such as modules, generator functions, for…of, and tagged template
+    // literals. Because they’re too varied to detect, pass all CoffeeScript
+    // compiler output through the Babel compiler.
+    const doubleRoastedCoffee =
+      this.babelCompiler.processOneFileForTarget(inputFile, output.js);
```

Comment:
If you just mean convincing yourself that it makes a difference, you can `console.log({cacheDir})` and then inspect that directory to verify that compiled `.json` files are getting saved by the `BabelCompiler`. You could also put some timing code around `this.babelCompiler.processOneFileForTarget(...)` to make sure it gets faster after the first compile.

If you mean programmatically testing that it works, then I'm not sure what the best strategy is. It's tricky because caching is supposed to be a transparent optimization, so it can be hard to detect/verify from the outside.